src/docs/Foundations/Typography/HeadingStyles/code.mdx

```diff
@@ -3,81 +3,124 @@ title: 'Code'
-> Headings atoms sizes change at the 'M' viewport, witch correspond to tablet sizes.
+### Sizes
 
-#### Import Headings atoms
+There is 3 Headings sizes available. Each heading size rescale it's font-size at the **M** and **L** viewports.
+Use the size modifiers in combination to the `mt-heading` class :
 
-```scss
-@import 'settings-tools/_all-settings';
+- **Small** : `mt-heading mt-heading--s`
+- **Medium** : `mt-heading mt-heading--m`
+- **Large** : `mt-heading mt-heading--l`
 
-@include import-font-families();
+<Preview path="typographyHeading-sizes" />
 
-@import 'typography/_t.headings';
-```
+| Name          | < screen-m (680px)   | >= screen-m (680px)  | screen-l (1024px)    |
+| ------------- | -------------------- | -------------------- | -------------------- |
+| **Heading S** | 07 / 23px (1.4375mu) | 08 / 28px (1.75mu)   | 09 / 34px (2.125mu)  |
+| **Heading M** | 08 / 28px (1.75mu)   | 09 / 34px (2.125mu)  | 10 / 41px (2.4625mu) |
+| **Heading L** | 09 / 34px (2.125mu)  | 10 / 41px (2.4625mu) | 11 / 49px (3.0625mu) |
 
-#### usage
 
-```html
-<p class="mt-heading-m">Lorem ipsum...</p>
-<p class="mt-heading-m mt-heading-m--semi-bold">Lorem ipsum...</p>
-<p class="mt-heading-m mt-heading-m--line-height-s">Lorem ipsum...</p>
-<p class="mt-heading-m mt-heading-m--line-height-s mt-heading-m--semi-bold">
-  Lorem ipsum...
-</p>
-```
+### Font-weights
+
+By default, the font-weight of all headings are set to `regular`.
+A semi-bold variation is available by applying the `mt-heading--semi-bold` modifier.
+
+<Preview path="typographyHeading-weight" />
+
+
+### Colors
+
+There is 2 text colors available for the Headings :
+
+- **Default** : the current color inherited from the context
+- **Lightest** : `mt-heading--lightest` (`$color-text-lightest` / `#fff` ) for use within a dark backgrounds.
+
+<Highlight theme="warning" title="Keep in mind">
+
+**The headings color should be the darkest color** (`$color-text-darkest` / `#000`) in most cases.
+We do not define the default color for you to be able to override it in very specific edge cases.
+
+</Highlight>
+
+<Preview path="typographyHeading-colors" />
+
+### Underlines
+
+Add underlines by applying the `mt-heading--underline` modifier.
+
+There is 4 underline color options for you to use, 2 for use within light backgrounds and 3 for use withing dark backgrounds :
+
+#### Hover both light and dark background
+
+- **Default** : `mt-heading--underline` will be `$color-primary-01-500` (`#78be20`)
+
+#### Hover light background only
+
+- **Primary 02** : `mt-heading--underline mt-heading--line-primary-02` will be `$color-primary-02-500` (`#78be20`)
+
+#### Hover dark background only
+
+- **Reverted** : `mt-heading--underline mt-heading--line-reverted` will be `$color-primary-01-200`
+- **Primary 02 reverted** : `mt-heading--underline mt-heading--line-primary-02-reverted` will be `$color-primary-02-200`
```

Comment:
This whole part is a little hard to understand. Can we simplify it into two categories:

### Underlines that could be used with a light background
- Default
- Primary 02

### Underlines that could be used with a dark background
- Primary 02
- Reverted
- Primary 02 reverted

Reply:
On the other hand, can we rename the term `reverted` by a more meaningful term?

Reply:
yep decided in pooling, we will use the color values (`mt-headings--line-primary-01-200`)